framework/src/main/java/org/fulib/fx/controller/internal/ReflectionSidecar.java

```diff
@@ -101,7 +101,7 @@ private void callMethodsWithAnnotation(
      */
     private void fillParametersIntoFields(@NotNull Object instance, @NotNull Map<@NotNull String, @Nullable Object> parameters) {
         // Fill the parameters into fields annotated with @Param
-        for (Field field : Reflection.getAllFieldsWithAnnotation(instance.getClass(), Param.class).toList()) {
+        for (Field field : ReflectionUtil.getAllNonPrivateFieldsOrThrow(instance.getClass(), Param.class).toList()) {
```

Comment:
Just realized this could be changed to stream.forEach to avoid the intermediate list